instrumentation/spring/spring-scheduling-3.1/javaagent/src/test/java/SpringSchedulingTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import spring.component.IntervalTask;
+import spring.component.OneTimeTask;
+import spring.component.TaskWithError;
+import spring.component.TriggerTask;
+import spring.config.EnhancedClassTaskConfig;
+import spring.config.IntervalTaskConfig;
+import spring.config.LambdaTaskConfig;
+import spring.config.OneTimeTaskConfig;
+import spring.config.TaskWithErrorConfig;
+import spring.config.TriggerTaskConfig;
+import spring.service.LambdaTaskConfigurer;
+
+class SpringSchedulingTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  void scheduleOneTimeTest() throws InterruptedException {
+    try (AnnotationConfigApplicationContext context =
+        new AnnotationConfigApplicationContext(OneTimeTaskConfig.class)) {
+      OneTimeTask task = context.getBean(OneTimeTask.class);
+      task.blockUntilExecute();
+
+      assertThat(task).isNotNull();
+      assertThat(testing.waitForTraces(0)).isEmpty();
+    }
+  }
+
+  @Test
+  void scheduleCronExpressionTest() throws InterruptedException {
+    try (AnnotationConfigApplicationContext context =
+        new AnnotationConfigApplicationContext(TriggerTaskConfig.class)) {
+      TriggerTask task = context.getBean(TriggerTask.class);
+      task.blockUntilExecute();
+
+      assertThat(task).isNotNull();
+      testing.waitAndAssertTraces(
+          trace ->
+              trace
+                  .hasSize(1)
```

Comment:
usually we don't use `hasSize` as `hasSpansSatisfyingExactly` also verifies the size

Reply:
Removed superfluous `hasSize` assertion